### PR TITLE
fix(kit): `InputPhone` should erase value on blur if it's just country code

### LINKED ISSUE #6199: 🐞 - `InputPhone` should erase value on blur if it's just country code
### Playground Link

_No response_

### Description

We should clear the input on blur when there is no value other than forced country code.

We also need to have empty string in form control while value is only the country code to allow optional form control to pass validation as empty.

### Angular version

_No response_

### Taiga UI version

_No response_

### Which browsers have you used?

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #6199
Relates #6060

### CODE REVIEW

File: projects/kit/components/input-phone/input-phone.component.ts [hunk 1]
```diff
@@ -181,6 +181,8 @@ export class TuiInputPhoneComponent
         if (this.nativeValue === this.nonRemovablePrefix || this.isTextValue) {
```
Comment (waterplea) [inline projects/kit/components/input-phone/input-phone.component.ts:181]: Should it be `this.nativeValue.trim() === this.nonRemovablePrefix.trim()`? Is it possible one of them contains space at the end and the other does not?

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-phone/input-phone.component.ts [hunk 2]
```diff
@@ -181,6 +181,8 @@ export class TuiInputPhoneComponent
...
             this.updateSearch('');
             this.nativeValue = '';
+
+            return;
         }
 
         if (!active && !this.allowText && this.nativeFocusableElement) {
```
Comment (waterplea) [inline projects/kit/components/input-phone/input-phone.component.ts:188]: What does this branch do?

Comment (nsbarsukov) [reply projects/kit/components/input-phone/input-phone.component.ts:188]:
```
"+1 (123) 456-" => blur => "+1 (123) 456"
```

File: projects/kit/components/input-phone/input-phone.component.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/kit/components/input-phone/input-phone.component.ts:181]: >  Is it possible one of them contains space at the end and the other does not?

* `nonRemovablePrefix` always includes space at the end:
https://github.com/taiga-family/taiga-ui/blob/a67423fe82616925ebaeb912b627b9e66a6e3984/projects/kit/components/input-phone/input-phone.component.ts#L227-L229

* User can't delete space in `nativeValue`
https://github.com/taiga-family/taiga-ui/blob/a67423fe82616925ebaeb912b627b9e66a6e3984/projects/kit/components/input-phone/input-phone.component.ts#L269

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**